apps/src/applab/designElements/elementUtils.js

```diff
@@ -234,9 +234,16 @@ export function setDefaultBorderStyles(element, options = {}) {
  * @param {string} cssPaddingString value from element.style.padding
  */
 export function calculatePadding(cssPaddingString) {
+  //
+  // paddingRegEx matches a string with up to 4 pairs of numeric values
+  // with or without pixel suffixes (e.g. 0 or 2px)
+  //
   const paddingRegEx = /\s*?(\d*)(px)?\s*(\d*)?(px)?\s*(\d*)?(px)?\s*(\d*)?(px)?\s*?/i;
```

Comment:
Something to consider as we take this work out of experiment: This all seems to work well, but there may be a simpler solution.  For once, I'll recommend against using regex. The first eight lines of this function collapse:

```js
const paddingValues = cssPaddingString.split(/\s+/).map(part => parseInt(part, 10))
```

This also lacks some flexibility. I suspect it's sufficient to cover existing App Lab element CSS, but it'd be more future-proof to write this in a way that supported CSS syntax more generally (in particular, specifying padding in units other than `px`).

Is there a reason we can't use [`Window.getComputedStyle`](https://developer.mozilla.org/en-US/docs/Web/API/Window/getComputedStyle) to support more unit types and remove all the special knowledge of CSS padding syntax from this code?

```js
export function calculatePadding(element) {
  const style = window.getComputedStyle(element);
  return {
    horizontalPadding: style.getPropertyValue('padding-left') + style.getPropertyValue('padding-right'),
    verticalPadding: style.getPropertyValue('padding-top') + style.getPropertyValue('padding-bottom')
  };
}
```

Reply:
Chrome & Safari don't support `getComputedStyle()` on detached DOM elements, which we rely on during drag/drop.

I'm happy to get rid of the regex and use your suggestion.

Added comments to explain that we will ignore other units like `em` and an explanation for why we can't use `getComputedStyle()`

Reply:
Bummer.

That feels like a solvable problem.  It seems like element padding shouldn't be changing during drag/drop so we could precompute it at the start of that operation.  Also interested to learn more about how we rely on detached elements during drag/drop: The element is still visible while dragging so I assumed it was still in the DOM.

I'm okay with not solving as part of this PR (unrelated complexity), but reimplementing CSS parsing bothers me and I'd like us to clean this up in the near future.

Reply:
This is specifically for dragging from the toolbox into the design mode screen (as part of the create path). I'm sure we attach it eventually, but the label sizing code runs when the element is first created. I agree that it would be nice to simplify this.